core/src/test/java/org/bitcoinj/core/FullBlockTestGenerator.java

```diff
@@ -1131,7 +1131,8 @@ public boolean add(Rule element) {
         {
             Transaction tx = new Transaction(params);
             tx.addOutput(new TransactionOutput(tx, ZERO, new byte[] {}));
-            b58.getSpendableOutput().outpoint.setIndex(42);
+            TransactionOutPoint outPoint = b58.getSpendableOutput().outpoint;
+            b58.getSpendableOutput().outpoint = new TransactionOutPoint(42, outPoint.getHash());;
```

Comment:
Have you checked if the other fields need copying?

Reply:
Well, I ran the tests and they worked. I'm going to change this line to produce a "connected" outPoint by using a different constructor -- one that _should_ result in all the same fields.

Reply:
Changes force-pushed.